configs/server.toml

```diff
@@ -1,122 +1,337 @@
+# "disabled" means messages are kept indefinitely.
+# A time value in seconds determines the lifespan of messages.
 message_expiry = "disabled"
+
+# Maximum size of a topic, e.g., "10 GB".
 max_topic_size = "10 GB"
 
+# Encryption configuration
 [system.encryption]
+# Determines whether server-side data encryption is enabled (boolean).
+# `true` enables encryption for stored data using AES-256-GCM.
+# `false` means data is stored without encryption.
 enabled = false
+
+# The encryption key used when encryption is enabled (string).
+# Should be a 32-byte length key, provided as a base64 encoded string.
+# This key is required and used only if encryption is enabled.
 key = ""
 
+# Compression configuration
 [system.compression]
+# Allows overriding the default compression algorithm per data segment (boolean).
+# `true` permits different compression algorithms for individual segments.
+# `false` means all data segments use the default compression algorithm.
 allow_override = false
+
+# The default compression algorithm used for data storage (string).
+# "none" indicates no compression, other values can specify different algorithms.
 default_algorithm = "none"
 
+# Stream configuration
 [system.stream]
+# Path for storing stream-related data (string).
+# Specifies the directory where stream data is stored, relative to `system.path`.
 path = "streams"
 
+# Topic configuration
 [system.topic]
+# Path for storing topic-related data (string).
+# Specifies the directory where topic data is stored, relative to `stream.path`.
 path = "topics"
 
+# Partition configuration
 [system.partition]
+# Path for storing partition-related data (string).
+# Specifies the directory where partition data is stored, relative to `topic.path`.
 path = "partitions"
+
+# Controls whether message deduplication is enabled (boolean).
+# `true` activates deduplication, ignoring messages with duplicate IDs.
+# `false` treats each message as unique, even if IDs are duplicated.
 deduplicate_messages = false
+
+# Determines whether to enforce file synchronization on partition updates (boolean).
+# `true` ensures immediate writing of data to disk for durability.
+# `false` allows the OS to manage write operations, which can improve performance.
 enforce_fsync = false
+
+# Enables checksum validation for data integrity (boolean).
+# `true` activates CRC checks when loading data, guarding against corruption.
+# `false` skips these checks for faster loading at the risk of undetected corruption.
 validate_checksum = false
+
+# The threshold of buffered messages before triggering a save to disk (integer).
+# Specifies how many messages accumulate before persisting to storage.
+# Adjusting this can balance between write performance and data durability.
 messages_required_to_save = 10_000
 
+# Segment configuration
 [system.segment]
+# Defines the maximum size of a storage segment in bytes (integer).
+# When a segment reaches this size, a new segment is created for subsequent data.
```

Comment:
please mention that it's soft limit, maximum segment size is actually this value + sizeof single message

Reply:
Updated, I also put the explanation behind in a form of example